clpy/backend/opencl/env.pyx

```diff
@@ -72,6 +73,33 @@ cdef void check_device_version(cl_device_id device, required_version) except *:
     if not interpret_versionstr(versionstr) >= required_version:
         raise RuntimeError("Device's OpenCL version must be >= 1.2")
 
+cdef cl_bool check_device_extension(cl_device_id device,
+                                    str extension) except *:
+    cdef size_t param_value_size
+    api.GetDeviceInfo(
+        device,
+        CL_DEVICE_EXTENSIONS,
+        0,
+        NULL,
+        &param_value_size)
+
+    cdef cython_array extensions_buffer =\
+        cython_array(shape=(param_value_size,),
+                     itemsize=sizeof(char),
+                     format='b')
+    api.GetDeviceInfo(
+        device,
+        CL_DEVICE_EXTENSIONS,
+        param_value_size,
+        <void*>extensions_buffer.data,
+        &param_value_size)
+
+    extensions =\
+        extensions_buffer.data[:param_value_size]\
+        .decode(locale.getpreferredencoding())
+
+    return CL_TRUE if extension in extensions else CL_FALSE
```

Comment:
This could be a bug when `extensions` doesn't contain `extension` but "XXXXXXextensionYYYYYY". You should separate the returned string and match them individually.

Reply:
I fixed it on aaf590c . Thanks.